packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -107,6 +110,91 @@ export class FromCloudFormation {
     return ret;
   }
 
+  public static parseCreationPolicy(policy: any): CfnCreationPolicy | undefined {
+    if (typeof policy !== 'object') { return undefined; }
+
+    // change simple JS values to their CDK equivalents
+    policy = FromCloudFormation.parseValue(policy);
+
+    return undefinedIfAllValuesAreEmpty({
+      autoScalingCreationPolicy: parseAutoScalingCreationPolicy(policy.AutoScalingCreationPolicy),
+      resourceSignal: parseResourceSignal(policy.ResourceSignal),
+    });
+
+    function parseAutoScalingCreationPolicy(p: any): CfnResourceAutoScalingCreationPolicy | undefined {
+      if (typeof p !== 'object') { return undefined; }
+
+      return undefinedIfAllValuesAreEmpty({
+        minSuccessfulInstancesPercent: FromCloudFormation.getNumber(p.MinSuccessfulInstancesPercent),
+      });
```

Comment:
Maybe a style thing but why all these function-style code when it's not even invoked more than once.

Wouldn't it be simpler to just put this inline.

```ts
if (typeof policy.AutoScalingCreationPolicy !== 'object') { return undefined; }
const autoScalingPolicy = undefinedIfAllValuesAreEmpty({
  minSuccessfulInstancesPercent: FromCloudFormation.getNumber(p.MinSuccessfulInstancesPercent),
});
```

Fine if you want to leave it as is.

Reply:
What you posted is not actually equivalent, because you're missing the `if (typeof p !== 'object')` condition.

I think your example code  with those conditions included would be a lot messier than what we have there now.